pandas/core/indexes/base.py

```diff
@@ -543,10 +548,13 @@ def __new__(
             if len(data) == 0:
                 # unlike Series, we default to object dtype:
                 data = np.array(data, dtype=object)
-
-            if len(data) and isinstance(data[0], tuple):
+            elif isinstance(data[0], tuple):
                 # Ensure we get 1-D array of tuples instead of 2D array.
                 data = com.asarray_tuplesafe(data, dtype=_dtype_obj)
+            elif dtype and dtype.kind == "S":
+                # GH#50127 we update data to a np.array with the correct dtype.
+                data = np.array(data, dtype=object, copy=copy)
```

Comment:
after doing this can we set copy=False to ensure no subsequent copies

Reply:
Done.